### ISSUE TITLE
Feature request: Disable auto-fetch globally

### ISSUE THREAD

**Issue by svitalsky** [zielu/GitToolBox#186]
Could you please please please add an option to disable auto-fetch globally? I mean any auto-fetch, both scheduled and on branch change.

**Comment by zielu** [comment]
I'm not keen on doing configuration overrides between Global / Project config. 

If you don't want to remember to disable it when creating new project you can configure it in File / Other Settings / Preferences for new project

**Comment by zielu** [comment]
@svitalsky I'm curious what is the use case that makes you want do disable it ?

**Comment by svitalsky** [comment]
> I'm not keen on doing configuration overrides between Global / Project config.
> 
But you are actually doing it already, don't you—you have global setting (auto fetch on) and possibility to override it in project settings. I'm only suggesting that a user may have a chance to choose what that global setting would be.

> If you don't want to remember to disable it when creating new project you can configure it in File / Other Settings / Preferences for new project

That doesn't work for older projects I've been working on long before installing GItToolBox.

**Comment by svitalsky** [comment]
> @svitalsky I'm curious what is the use case that makes you want do disable it ?

My basic everyday workflow. Being an old Unix guy, I work whenever possible from command line. That includes how I control Git, I do everything from command line there (OK, I occasionally discard a hunk from Sublime Merge). All the GUI-like tools (I've bought Sublime Merge license, I use Meld for visual diffs sometimes and of course Idea, including recently your plugin) are—as far as Git is concerned—for nicer displaying of various information, i.e. with your plugin branch name and blame for the line, with Sublime Merge branch tree for merging and rebasing, with Meld those diffs &c.. I do not want anything to go auto.

**Comment by panda2025** [comment]
[https://github.com/panda2025/GitToolBox/releases](url)
here is disable auto-fetch

```java
internal data class GitToolBoxConfigPrj(
  var autoFetch: Boolean = false, //set default false
  var autoFetchIntervalMinutes: Int = AutoFetchParams.DEFAULT_INTERVAL_MINUTES,
  @Deprecated("Since 192.3.1") var autoFetchExclusions: List<String> = ArrayList(),
  var autoFetchExclusionConfigs: List<AutoFetchExclusionConfig> = ArrayList(),
  var autoFetchOnBranchSwitch: Boolean = false, //set default false
  var commitDialogCompletion: Boolean = true,
  var completionConfigs: List<CommitCompletionConfig> = arrayListOf(CommitCompletionConfig()),
  var referencePointForStatus: ReferencePointForStatusConfig = ReferencePointForStatusConfig(),
  var commitMessageValidation: Boolean = false,
  var commitMessageValidationRegex: String = "(?:fix|chore|docs|feat|refactor|style|test)(?:\\(.*\\))?: [A-Z].*\\s#\\d+"
) 
```

**Comment by zielu** [comment]
@panda2025, what about people that rely on default being enabled ? For then auto-fretch will stop working and they will have to enable for each project manually.

**Comment by zielu** [comment]
I have an idea how this could be forced at ide level config and not mess up existing configs. Prototype on plugin EAP channel should be ready before or after weekend

**Comment by zielu** [comment]
There will be an additional config entry in settings "GitToolBox Overrides" that will allow to specify both auto fetch related toggles to be overridden.
Overrides will be applied once on project open. After override is applied and project config is changed manually later override will not be applied the second time.

**Comment by zielu** [comment]
Prototype released to EAP channel as 193.4.0.

There is one change config is under "GitToolBox Global" > Extras.

Switching to EAP channel is described here: https://github.com/zielu/GitToolBox#eap-builds

**Comment by zielu** [comment]
Done in 193.4.2
